src/screens/registerStudent/hooks/useRegister.ts

```diff
@@ -0,0 +1,248 @@
+import { SelectChangeEvent } from '@mui/material/Select';
+import { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import useCourseRequest from '../../../service/requests/useCourseRequest';
+import useStudentRegisterRequest from '../../../service/requests/useStudentRegisterRequest';
+import { SCREENS } from '../../../utils/screens';
+import {
+  validateConfirmPassword,
+  validateEmail,
+  validateEnrollment,
+  validateName,
+  validatePassword,
+} from '../../../utils/validateFields';
+
+const useRegister = () => {
+  const navigate = useNavigate();
+
+  const { coursesFetch, data: courses } = useCourseRequest();
+
+  const { isLoading, isSuccess, registerStudent, error } =
+    useStudentRegisterRequest();
+
+  const [step, setStep] = useState(0);
+  const [courseId, setCourseId] = useState(0);
+  const [course, setCourse] = useState('Selecione um curso');
+  const [name, setName] = useState('');
+  const [enrollment, setEnrollment] = useState('');
+  const [description, setDescription] = useState('');
+  const [email, setEmail] = useState('');
+  const [contactEmail, setContactEmail] = useState('');
+  const [linkedin, setLinkedin] = useState('');
+  const [whatsapp, setWhatsapp] = useState('');
+  const [password, setPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [showPassword, setShowPassword] = useState(false);
+  const [nameError, setNameError] = useState('');
+  const [enrollmentError, setEnrollmentError] = useState('');
+  const [emailError, setEmailError] = useState('');
+  const [passwordError, setPasswordError] = useState('');
+  const [confirmPasswordError, setConfirmPasswordError] = useState('');
+
+  const handleContinueClick = (e: React.SyntheticEvent<EventTarget>) => {
+    let currentError;
+
+    e.preventDefault();
+    setNameError('');
+    setConfirmPasswordError('');
+    setPasswordError('');
+    setEmailError('');
+
+    currentError = validateName(name);
+    if (currentError !== '') {
+      return setNameError(currentError);
+    }
+
+    currentError = validateEmail(email);
+    if (currentError !== '') {
+      return setEmailError(currentError);
+    }
+
+    currentError = validatePassword(name, password);
+    if (currentError !== '') {
+      return setPasswordError(currentError);
+    }
+
+    currentError = validateConfirmPassword(password, confirmPassword);
+    if (currentError !== '') {
+      return setConfirmPasswordError(currentError);
+    }
+
+    handleStepChange(1);
+  };
+
+  const handleStepChange = (increment: number) => {
+    setStep((previousStep) => previousStep + increment);
+  };
+
+  const handleBackClick = () => {
+    handleStepChange(-1);
+  };
+
+  const handleSecondContinueClick = (e: React.SyntheticEvent<EventTarget>) => {
+    e.preventDefault();
+    setEnrollmentError('');
+
+    const currentError = validateEnrollment(enrollment);
+    if (currentError !== '') {
```

Comment:
Melhor fazer assim para ficar mais genérico:
```suggestion
    if (!currentError) {
```

Funciona da mesma forma porque o valor booleano de uma string vazia é `false`.

Reply:
Aderi aqui mor, mas no caso é só `if (currentError) {` pq eu quero que ele entre no trecho se ela tá preenchida